<?xml version="1.0" ?>
<launch xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
        xsi:noNamespaceSchemaLocation="https://gist.githubusercontent.com/nalt/dfa2abc9d2e3ae4feb82ca5608090387/raw/roslaunch.xsd">
    <!-- Arguments -->
    <arg name="log_level" default="INFO" />
    <arg name="hardware_version" default="ned" />
    <arg name="simulation_mode" default="false" />
    <arg name="gazebo" default="false" />
    <arg name="client_stream_topic" value="~client_stream" />


    <include file="$(find niryo_robot_vision)/launch/vision_node_base.launch.xml"> 
        <arg name="hardware_version" value="$(arg hardware_version)" />
        <arg name="simulation_mode" value="$(arg simulation_mode)" />
    </include>


    <node name="niryo_robot_vision" pkg="niryo_robot_vision" type="vision_node.py" output="screen">
        <param name="log_level" type="string" value="$(arg log_level)" />
        <param name="hardware_version" value="$(arg hardware_version)" />
        <param name="simulation_mode" type="bool" value="$(arg simulation_mode)" />

        <param name="client_stream_topic" value="~client_stream" unless="$(arg gazebo)" />
        <param name="client_stream_topic" value="/camera/image_raw/compressed" if="$(arg gazebo)" />
    </node>
</launch>
